### ISSUE TITLE
Allowing contact through "Do not Disturb" / Focus

### ISSUE THREAD

**Issue by aneillans** [TelegramMessenger/Telegram-iOS#1160]
It would be really good to get "full" integration on iOS to allow whitelisting an individual contact to break through on DnD / Focus modes in Telegram, much like things like WhatsApp allows. I believe this can be done by integrating into the address book and notifications further ?

**Comment by ali-fareed** [comment]
Doesn't it work currently?

**Comment by aneillans** [comment]
Nope; you have to exclude the entire app which means you then have always mute full conversations. Which is less than ideal.

The ideal would be the Telegram contact type to persist properly on the contact record (it doesn't just reverts to other on current iOS build, 16.6), and for this to be correctly handle in notification filtering.

**Comment by ali-fareed** [comment]
Adding regular phonebook contacts to focus exceptions works just fine for me. Maybe yours are not stored using the normalized (international) phone format?

Here is when the OS contact id is resolved: https://github.com/TelegramMessenger/Telegram-iOS/blob/master/Telegram/NotificationService/Sources/NotificationService.swift#L1852

**Comment by aneillans** [comment]
I wonder if that's the key thing - a lot of my telegram contacts don't share their numbers on the platform (even where I have it) due to security issues they perceive on the platform itself / with others - therefor I bet the resolution is not working, as it only looks for the number and not the other service ?

**Comment by JavierSegoviaCordoba** [comment]
Up
